HelpSource/Guides/Tour-of-Special-Functions.schelp

```diff
@@ -0,0 +1,1065 @@
+
+// generate colors for each plot
+genCol = { |num, offset=(rrand(0,1.0)), range=(rrand(0.15,1.3))|
+	num.collect{|i| Color.hsv(((i/(max(num,1.0001)-1)) * range + offset).wrap(0,1), 0.6, 1, 1)}
+};
+// generate a vertical layout for the plot and its text
+getPlotLay = { |plot, text|
+	VLayout(plot, StaticText().string_(text).maxHeight_(50).align_(\center), 10);
+};
+
+~multiPlot = { |func, data, txt, min, max|
+	var res, params, samps, pltr;
+	var gridcol = Color.gray.alpha_(0.3);
+
+	res = if (data.shape.size == 1) { // 1D data
+		params = [0];
+		samps = data;
+		samps.collect({ |smp|
+			func.applyTo(smp)
+		}).select({|me| me.notNil})
+	}
+	{ // 2D data
+		#params, samps = data;
+		txt = txt ++ params.asString;
+		params.collect({ |param, i|
+			samps.collect({ |smp|
+				func.applyTo(param, smp, i)
+			}).select({|me| me.notNil})
+		});
+	};
+
+	pltr = res.plot( txt, minval: min, maxval: max).superpose_(true);
+
+	pltr.plots[0]
+	.plotColor_(genCol.(params.size, rrand(0,1.0)))
+	.domainSpec_([samps.minItem, samps.maxItem].asSpec)
+	.gridColorX_(gridcol).gridColorY_(gridcol)
+	;
+
+	pltr
+};
+
+~layOutPlots = { |plotters, title="", numCols|
+	var cols, hl, plotlay, colbut, plots, wscl;
+	var vl = VLayout();
+	var sb = Window.screenBounds;
+
+	cols = numCols ?? {plotters.size.sqrt.ceil.asInt};
+	wscl = cols.linlin(1, 4, 0.5, 0.9);
+	w = Window(title,
+		Rect().width_(sb.width*wscl).height_(sb.height).center_(sb.center)
+	).view.layout_(vl).front;
+
+	colbut = Button().states_([["Light Plots"],["Dark Plots"]]).maxSize_(Size(80,15));
+	vl.add(HLayout(nil, colbut));
+
+	plotters.do{ |pltr, i|
+		var plot = pltr.parent.view.children[0];
+		if( i%(cols) == 0 ) {vl.add(hl = HLayout())};
+		plotlay = getPlotLay.(plot, pltr.name.asString);
+		hl.add(plotlay);
+		pltr.parent.close;
+	};
+	if((plotters.size)%cols > 0) {
+		((cols-(plotters.size)%cols)).do{hl.add(StaticText().string_(""))}
+	};
+
+	colbut.action_({ |but|
+		var switch = but.value.asBoolean;
+		plotters.do{ |pltr|
+			pltr.plots[0]
+			.backgroundColor_(
+				switch.if({Color.white.alpha_(1)}, {Color(*0.1.dup(3)).alpha_(0.95)})
+			)
+			.fontColor_(
+				switch.if({Color.gray},{Color.red});
+			);
+		};
+		w.refresh;
+	});
+	colbut.value_(0).doAction; // start with dark plots
+};
+)
+::
+
+SECTION:: Number Series
+
+CODE::
+/* Bernouli Numbers */
+
+//   Odd numbered Bernoulli numbers are zero, except B1
+//   which is -1/2. Post even Bernouli numbers:
+(0..31).do{|i| postf("% %\n", i*2, bernouliB2n(i))}
+
+
+/* Tangent numbers */
+
+(0..15).do{|i| postf("% %\n", i, tangentT2n(i))}
+::
+
+
+SECTION:: Gamma Functions
+
+SUBSECTION:: Gamma, Log Gamma, Digamma, Trigamma, Polygamma
+
+Be sure to define code::~layOutPlots:: and code::~multiPlot:: at the link::#Before you get started...#top of the page::.
+CODE::
+(
+var plotSpecs = [
+	[   // tgamma(z)
+		{|z| // make sure z isn't a negative integer
+			if (((z<=0) and: ((z % 1) ==0)).not) { tgamma(z) }
```

Comment:
for efficiency, this should be:
` 		if (((z<=0) and: { ((z % 1) ==0)).not }) { tgamma(z) } `
(it will inline, same the other cases below)

and in this case of course:

` 		if (((z<=0) and: { (z % 1) !=0) }) { tgamma(z) } `

Reply:
even though it won't inline as nicely, I'd also tend to make a function:

```supercollider
var call = { |x, selector|  // make sure x isn't a negative integer
if((x <=0) and: { (x % 1) !=0) }) { x.perform(selector) }
};
```

and then just call this function further down, e.g.

```supercollider
{|dz| call.value(dz, \tgamma1pm1) }
```